src/test/e2e.test.ts

```
@@ -55,7 +55,24 @@ describe('E2E', () => {
         await execFile('git', ['commit', '-m', `first migration\n\n[merkel up ${uuid}]`], { env: { PATH } })
         let status = await getStatus(adapter, await getHead())
         assert.equal(status.newCommits.length, 1)
-        await status.executePendingTasks('migrations', adapter, SILENT_LOGGER)
+        assert.equal(status.newCommits[0].tasks.length, 1)
+        // add pending task
+        await adapter.beginMigrationTask({ ...status.newCommits[0].tasks[0], head: await getHead() } as Task)
+        let shouldHaveFinished = false
+        // begin execution, this should wait
+        const promise = new Promise<void>(async (resolve, reject) => {
```

This swallows exceptions, you can make an async IIFE